cartridge/ddl-manager.lua

```diff
@@ -0,0 +1,128 @@
+local log = require('log')
+local ddl = require('ddl')
+local yaml = require('yaml')
+local checks = require('checks')
+local errors = require('errors')
+
+local failover = require('cartridge.failover')
+
+local SetSchemaError = errors.new_class('SetSchemaError')
+local CheckSchemaError = errors.new_class('CheckSchemaError')
+local DecodeYamlError = errors.new_class('DecodeYamlError')
+
+local _section_name = 'schema.yml'
+local _example_schema = [[## Example:
+#
+# spaces:
+#   customer:
+#     engine: memtx
+#     is_local: false
+#     temporary: false
+#     sharding_key: [customer_id]
+#     format:
+#       - {name: customer_id, type: unsigned, is_nullable: false}
+#       - {name: bucket_id, type: unsigned, is_nullable: false}
+#       - {name: fullname, type: string, is_nullable: false}
+#     indexes:
+#     - name: customer_id
+#       unique: true
+#       type: TREE
+#       parts:
+#         - {path: customer_id, type: unsigned, is_nullable: false}
+#
+#     - name: bucket_id
+#       unique: false
+#       type: TREE
+#       parts:
+#         - {path: bucket_id, type: unsigned, is_nullable: false}
+#
+#     - name: fullname
+#       unique: true
+#       type: TREE
+#       parts:
+#         - {path: fullname, type: string, is_nullable: false}
+]]
+
+local function _from_yaml(schema_yml)
+    if type(schema_yml) ~= 'string' then
+        return nil, CheckSchemaError:new(
+            'Section %q must be a string, got %s',
+            _section_name, type(schema_yml)
+        )
+    end
+
+    return DecodeYamlError:pcall(yaml.decode, schema_yml)
+end
+
+local function apply_schema_from_yaml(conf, opts)
+    checks('table', {is_master = 'boolean'})
+
+    if not opts.is_master then
+        return true
+    end
+
+    if conf[_section_name] == nil then
```

Comment:
I think the section name is redundant here. The corresponding section as-is will be passed from Cartridge conf applier.

Reply:
Decided to leave as is